### DISCUSSION TITLE
RSS Feed Generation

### DISCUSSION THREAD

**Discussion by prncevince** [quarto-dev/quarto-cli#7280] [Feature Requests]
### Description

Quarto Website Document Listing Feeds Reference:

- https://quarto.org/docs/websites/website-listings.html#feeds

It appears that the following option includes a large portion of the HTML webpage (everything in `<main>` after `<header>`) in the `<item><description>` tag/node of an RSS feed document XML format document:

```yaml
listing:
  feed:
    type: full
```

Additionally, it appears that the following option includes the contents of the first `<p>` encountered after `<main><header>` HTML tag within `<item><description>` of the XML:

```yaml
listing:
  feed:
    type: partial
```

This can significantly reduce the amount of HTML content pushed to the RSS feed XML document. 

Additionally, options like YAML header metadata `description`s for listing item pages do not get pushed to the RSS feed file in either of these options. 

I'm new to the RSS feed game, but I believe that storing rich content (e.g. iframes from widgets) may be overkill. This can produce very large RSS feeds that popular RSS readers (e.g. Feedly), do not actually parse. Additionally, if the goal is to generate additional custom content automatically from the RSS feed (e.g. a newsletter), this type of content may not be desired.

I guess that one could run a post processing text parsing script on their Quarto project to remove such content from the `<description><![CDATA[[ ...]]></description>` XML tag using `project:post-render`. 

Or maybe more options could be built-in to base Quarto as extra `listing:feed:type` options? 

Or possible, a `listing:feed:remove` field can be added to selectively remove rich content based on scraping search matches? 

e.g. `[no-iframes, no-social-embeds, etc.]`

Thoughts? Thanks!

**Comment by prncevince** [comment]
Possibly even a custom CSS style selector to be specified to select & remove DOM content from the RSS feed? 

This would probably provide the _most_ flexibility.

**Comment by dragonstyle** [comment]
We already do a decent amount of processing to the HTML to make it more reader friendly, I think this is just an oversight and something we really should do automatically. I'm opening an issue to track that.

**Comment by mcanouil** [comment]
Closing in favour of:
- #7290
